index.js

```diff
@@ -396,6 +410,16 @@ function session(options) {
       });
     }
 
+    // check if session has been or is being destroyed
+    function isBeingDestroyed(req) {
+      return req.sessionID && req.session == null;
```

Comment:
Should this check that `unsetDestroy` is true here? That option is what makes the unsetting of `req.session` actually mean the session is to be destroyed, otherwise it does nothing.